src/ValidationState.php

```diff
@@ -25,40 +25,25 @@ class ValidationState implements ArrayAccess
      */
     public function __construct(array $verdicts)
     {
-        $this->verdicts = $verdicts;
+        $this->verdicts = new Verdicts($verdicts);
     }
 
     /**
      * @return bool
      */
     public function isValid(): bool
     {
-        foreach ($this->verdicts as $verdict) {
-            if (!$verdict->passes()) {
-                return false;
-            }
-        }
-
-        return true;
+        return count($this->verdicts->failing()) === 0;
```

Comment:
1: Literals should be on the left side of an expression, for the sake of redability.